share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -471,34 +471,50 @@ function(ExternalZephyrProject_Cmake)
 endfunction()
 
 # Usage:
-#   sysbuild_module_call(<hook> MODULES <modules> [IMAGES <images>] [EXTRA_ARGS <arguments>])
+#   sysbuild_module_call(<hook> MODULES <modules> [IMAGES <images>] [IMAGE <image>] [EXTRA_ARGS <arguments>])
 #
 # This function invokes the sysbuild hook provided as <hook> for <modules>.
 #
 # If `IMAGES` is passed, then the provided list of of images will be passed to
-# the hook.
+# the hook, if `IMAGE` is passed, this will be provided to the hook.
 #
 # `EXTRA_ARGS` can be used to pass extra arguments to the hook.
 #
 # Valid <hook> values:
-# PRE_CMAKE   : Invoke pre-CMake call for modules before CMake configure is invoked for images
-# POST_CMAKE  : Invoke post-CMake call for modules after CMake configure has been invoked for images
-# PRE_DOMAINS : Invoke pre-domains call for modules before creating domains yaml.
-# POST_DOMAINS: Invoke post-domains call for modules after creation of domains yaml.
+# PRE_CMAKE       : Invoke pre-CMake call for modules before CMake configure is invoked for images
+#                   (IMAGES variable provided)
```

Comment:
Technically it's the caller of `sysbuild_module_call(PRE_CMAKE ...)` which decides that `IMAGES` variable is provided and not this function.

If you want to have `IMAGES` a required argument for the `sysbuild_module_call(PRE_CMAKE ...)` call, then we should check that `IMAGES` is passed along with `PRE_CMAKE`.

This is mixing documentation of the function itself, with how sysbuild actually uses the function.

Same comment applies to the other hook descriptions.

I'm in favor of making the `IMAGES` or `IMAGE` arguments mandatory to reflect how we actually uses the hooks, which also means the text can reflect this requirement.

Reply:
have updated the text and made variables mandatory